backtrace-library/src/main/java/backtraceio/library/base/BacktraceBase.java

```diff
@@ -445,6 +479,72 @@ public boolean addBreadcrumb(String message, Map<String, Object> attributes, Bac
         return database.getBreadcrumbs().addBreadcrumb(message, attributes, type, level);
     }
 
+    /**
+     * Enable backtrace metrics support
+     * @param settings  Backtrace metrics settings
+     */
+    public void enableMetrics(BacktraceMetricsSettings settings) {
+        // For tracking crash-free sessions we need to add
+        // application.session and application.version to Backtrace attributes
+        String sessionId = UUID.randomUUID().toString();
+        this.attributes.put("application.session", sessionId);
+        try {
+            this.attributes.put("application.version", this.context.getPackageManager()
```

Comment:
I would recommend to use BacktraceAttributes method instead.

Reply:
+1

Reply:
This will add attributes directly to Backtrace custom report attributes (BacktraceBase.attributes). Do we still need to change this @konraddysput @vlussenburg ?